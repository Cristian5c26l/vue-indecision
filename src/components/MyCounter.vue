<template>
  <section>
    <h3>Counter: {{ counter }}</h3>
    <h3>Square: {{ squareCounter }}</h3>
  </section>

  <div>
    <button class="btn" @click="counter++">+1</button>
    <button class="btn" v-on:click="counter--">-1</button>
  </div>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter';

interface Props {
  value: number;
}

// Componente MyCounter debe recibir o requiere propiedad o propertie llamada "value" de tipo number
const props = defineProps<Props>();
const { counter, squareCounter } = useCounter(props.value);
</script>

<style scoped>
.btn {
  @apply p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2;
}
</style>
