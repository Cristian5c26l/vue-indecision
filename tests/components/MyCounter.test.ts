import { describe, test } from 'vitest';
import { mount } from '@vue/test-utils';
import MyCounter from '@/components/MyCounter.vue';

describe('<MyCounter />', () => {
  test('should match snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 50,
      },
    });

    // console.log(wrapper.html());

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should renders the counter value correctly and renders the counter square value correctly', () => {
    const value = 50;

    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });

    const [counterLabel, squareLabel] = wrapper.findAll('h3');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );

    expect(counterLabel.text()).toContain(`Counter: ${value}`);
    expect(squareLabel.text()).toContain(`Square: ${value * value}`);
  });

  test('should increments the counter when +1 button is clicked', async () => {
    const value = 50;

    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });

    const [counterLabel, squareLabel] = wrapper.findAll('h3');
    const btnIncrement = wrapper.find('button');

    await btnIncrement.trigger('click');

    expect(counterLabel.text()).toContain(`Counter: ${value + 1}`);
    expect(squareLabel.text()).toContain(`Square: ${(value + 1) * (value + 1)}`);
  });

  test('should decrements the counter when -1 button is clicked', async () => {
    const value = 50;

    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });

    const [counterLabel, squareLabel] = wrapper.findAll('h3');
    const [, btnDecrement] = wrapper.findAll('button');

    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');

    expect(counterLabel.text()).toContain(`Counter: ${value - 2}`);
    expect(squareLabel.text()).toContain(`Square: ${(value - 2) * (value - 2)}`);
  });
});
