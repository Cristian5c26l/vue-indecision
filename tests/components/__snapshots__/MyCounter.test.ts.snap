// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snapshot 1`] = `
"<section data-v-0352012b="">
  <h3 data-v-0352012b="">Counter: 50</h3>
  <h3 data-v-0352012b="" data-testid="square-label">Square: 2500</h3>
</section>
<div data-v-0352012b=""><button data-v-0352012b="" class="btn">+1</button><button data-v-0352012b="" class="btn">-1</button></div>"
`;
